Update DynamoDB Documentation

devdanco opened the pull request:
I split user manual (guide) into parts based on amazon S3, SNS etc to match how it should look like

- examples (this needs to be check. There are lot of changes)
- studio (this was copy / paste - took from file amazon-dynamodb-connector.adoc)
- xml-maven (this was copy / paste took from file amazon-dynamodb-connector.adoc)

devdanco committed:
Update DynamoDB Documentation

devdanco committed:
Add files via upload

devdanco committed:
Update amazon-dynamodb-connector-studio.adoc

devdanco committed:
Update amazon-dynamodb-connector.adoc

devdanco committed:
Create amazon-dynamodb-connector-xml-maven.adoc

devdanco committed:
Update amazon-dynamodb-connector.adoc

devdanco committed:
Add files via upload

devdanco committed:
Add files via upload

devdanco committed:
Create amazon-dynamodb-connector-examples.adoc

devdanco committed:
Add files via upload

devdanco committed:
Update amazon-dynamodb-connector-examples.adoc

devdanco committed:
Add files via upload

devdanco committed:
Add files via upload

devdanco committed:
Update amazon-dynamodb-connector-examples.adoc

dhirsch1 commented:
Michael, can you summarize what has changed?

devdanco committed:
Update amazon-dynamodb-connector.adoc

dhirsch1 commented:
Associated with https://www.mulesoft.org/jira/browse/DOCS-8088

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 commented:
Michael, thank you for putting the documentation in the template and for adding an example. I am updating the topics to reflect our current template wording (which has changed recently) and writing style.

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Draft of the Studio topic

dhirsch1 committed:
Fixed some minor errors

dhirsch1 committed:
Put information in the newest template for the landing page topic

dhirsch1 committed:
Interim commit for the Examples topic

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
++
+image::amazon/amazon-dynamodb-create-table-info.png[Create table info properties]
++
+. To add *Global Secondary Indexes*. add *variable* before the *Create Table* operation in this format:
```
```suggestion
. To add *Global Secondary Indexes*, add *variable* before the *Create Table* operation in this format:
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+image::amazon/amazon-dynamodb-create-table-logger-props.png[dynamodb create table logger]
++
+. Save and run the project as a Mule application: +
+In Package Explorer, right-click the project and click *Run As* > *Mule Application*.
```
Should be a substep

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+[[scan-operation]]
+== Perform a Scan
+
+This example shows how to use Amazon DynamoDB connector to perform an Amazon DynamoDB scan.
```
```suggestion
This example shows how to use Amazon DynamoDB Connector to perform an Amazon DynamoDB scan.
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+
+=== Flow for Performing a Scan
+
+The following screenshot shows the Studio flow for creating performing an Amazon DynamoDB scan:
```
creating or performing ?

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+=== Steps for Performing a Scan
+
+. Drag the Amazon DynamoDB Connector *Scan* operation next to the HTTP Listener component.
+. Only required field in *Scan* operation is *table name*
```
Line 356 is not really a step. I would just add it as info to the step on line 355, so:

. Drag the Amazon DynamoDB Connector *Scan* operation next to the HTTP Listener component. +
The only required field in the *Scan* operation is *table name*.

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+
+. Drag the Amazon DynamoDB Connector *Scan* operation next to the HTTP Listener component.
+. Only required field in *Scan* operation is *table name*
+. Add Transform Message after *Scan* operation in order to transform response to json
```
```suggestion
. Add Transform Message after the *Scan* operation to transform the response to JSON.
```
Probably should bold Transform Message.

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+image::amazon/amazon-dynamodb-to-json.png[Tranform response to json format]
++
+. Add a Logger component after the Amazon DynamoDB *Transform Message* operation to print the response that is generated by the *Scan* operation in the Mule Console.
+. The response should look like this (other items are removed for brevity)
```
```suggestion
. The response should look like this (other items are removed for brevity):
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+
+The DynamoDB AWS SDK paginates results by dividing them into specific pages. The 1MB limit applies to returned results. When this limit is exceeded, you must perform another scan to gather the rest of the data.
+
+To do perform a subsequent scans, apply the value of the *LastEvaluatedKey* field to the *ExclusiveStartkey* field. The *LastEvaluatedKey* value becomes null, the operation has completed all pages of data.
```
```suggestion
To perform subsequent scans, apply the value of the *LastEvaluatedKey* field to the *ExclusiveStartkey* field. The *LastEvaluatedKey* value becomes null, when the operation finishes scanning all the pages of data.
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,411 @@
+
+To do perform a subsequent scans, apply the value of the *LastEvaluatedKey* field to the *ExclusiveStartkey* field. The *LastEvaluatedKey* value becomes null, the operation has completed all pages of data.
+
+== Migration information
```
```suggestion
== Migration Information
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,142 @@
+To add an operation for Amazon DynamoDB Connector, follow these steps:
+
+. In the *Mule Palette* view, select *Amazon DynamoDB* and then select the desired operation.
+. Drag the operation onto the Studio canvas to the right of the input source.
```
to the right or on the right? I am never sure what the standard should be.

dhirsch1 replied:
I think "to the right" is OK because Kim never marked it.

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,142 @@
+|Try Default AWS Credentials Provider Chain | Set to true to obtain credentials from the AWS environment.
+|===
+
+The following screenshot shows...:
```
Looks like a placeholder. Don't forget to add the text.  :-)

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,142 @@
+|Region Endpoint | Select the topic region endpoint.
+|===
+
+The following screenshot shows..
```
shows..  ?

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,72 @@
+. Add the namespace for the connector to your XML flow.
+. Add dependency information to your Maven `pom.xml` file.
+
+In the Studio visual editor, when you drag a connector from the palette to the canvas, Studio automatically populates the XML code with the connector namespace and schema location.
```
did you remove the note?

dhirsch1 replied:
It is now a paragraph instead.

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,72 @@
+http://www.mulesoft.org/schema/mule/dynamodb/current/mule-dynamodb.xsd
+----
+
+This example shows how the namespace statements are placed in the mule XML block:
```
```suggestion
This example shows how the namespace statements are placed in the <mule> XML block:
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,72 @@
+
+== Add a POM File Dependency
+
+The `pom.xml` Maven file generated by Anypoint Studio contains dependencies for a Mule app. When you code a Mule app manually, include this XML snippet in your `pom.xml` file to enable access to this connector. The Mule runtime engine translates the `RELEASE` dependency keyword to the current version.
```
```suggestion
The `pom.xml` Maven file generated by Anypoint Studio contains dependencies for a Mule app. When you code a Mule app manually, include this XML snippet in your `pom.xml` file to enable access to this connector. Mule runtime engine translates the `RELEASE` dependency keyword to the current version.
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,64 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
-Anypoint Studio provides two ways to add the connector to your Studio project: 
+* Store and retrieve any amount of data
+* Serve any level of request traffic
+* Automatically spread the data and traffic for the table over a sufficient number of servers to handle the request capacity and the amount of data stored, while maintaining consistent and fast performance
```
```suggestion
* Automatically distribute the data and traffic for the table over a sufficient number of servers to handle the request capacity and the amount of data stored, while maintaining consistent and fast performance
```

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,64 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
-----
-<?xml version="1.0" encoding="UTF-8"?>
+Read the xref:amazon/amazon-dynamodb-connector-xml-maven.adoc[XML and Maven Support], and
+and xref:amazon/amazon-dynamodb-connector-examples.adoc[Examples] topics.
```
Is there an extra "and"?

valkyrie69 commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,64 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
 
+== Next Step
 
+After you complete the prerequisites and experimenting with templates and examples, you are ready to create a new app using Anypoint Studio.
```
experimenting or experiment?

dhirsch1 replied:
I will add "finish" as in "finish experimenting."

valkyrie69 reviewed (approved):
Minor comments

dhirsch1 committed:
Added Valkyries edits to the landing page and Studio topics

dhirsch1 committed:
Added screenshots and links to the topic. Fixed issues.

dhirsch1 committed:
One fix

dhirsch1 committed:
Prepared the Studio topic for a technical review

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Fixes and addition of some screenshots

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Added annotations, completed drafts of all topics

dhirsch1 committed:
Fixed screenshot name

dhirsch1 committed:
Removed accidental pasted info.

devdanco committed:
Fix create table example

I added globalSecondaryIndexes="#[vars.secondary]" to dynamodb:create-table. This is important to have this in XML when we creating secondary index.

dhirsch1 committed:
Fixes from edits

dhirsch1 committed:
A few tweaks before editorial review.

dhirsch1 committed:
Added Michal's edits

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
 
```
HERE

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
+For compatibility information and fixed issues, see the xref:release-notes::connector/amazon-dynamodb-connector-release-notes-mule-4.adoc[Amazon DynamoDB Connector Release Notes]
 
-Amazon DynamoDB is a fully managed NoSQL database service that provides fast and predictable performance with seamless scalability.
+== Common Use Cases For the Connector
```
```suggestion
== Common Use Cases for the Connector
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
-* From the Mule Palette view
+* Store and retrieve any amount of data
+* Serve any level of request traffic
+* Automatically distribute the data and traffic for the table over a sufficient number of servers to handle the request capacity and the amount of data stored. Amazon Dynamo DB Connector does this while maintaining consistent, fast performance.
```
Two solutions for the two-sentence structure:
Per our style ref....
```suggestion
* Automatically distribute the data and traffic for the table over a sufficient number of servers to manage the request capacity and the amount of data stored. +
Amazon Dynamo DB Connector does this while maintaining consistent, fast performance.
```
OR rewording for parallelism....
```suggestion
* Automatically distribute the data and traffic for the table over a sufficient number of servers to manage the request capacity and the amount of data stored this while maintaining consistent, fast performance
```

kbostrom replied:
One other suggestion, or really a question: is the third list item really sort of a recap of the first two?
If so, that might suggest other presentation possibilities.

kbostrom replied:
Just a note about "handle":
I think it's a bit vague, and not sure how translators "handle" it (see what I did there?).
I know that most sentence contexts imply the precise meaning of this term (in the previous sentence, for example, "handle" = "translate" or "localize" or similar).

However, I think it would be best if we simply use the precise meaning in context: manage, process, prioritize, control....

If you look this up, the first verb use is all about touching, followed by managing and then by carrying as merchandise (the noun entries are for, first, something to grip and, second, a nickname). Kinda' interesting.

https://www.thesaurus.com/browse/handle

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
-+
-image::amazon/amazon-dynamodb-use-case-config.png[DynamoDB use case configuration]
+* The Amazon DynamoDB API
+* Anypoint connectors
```
```suggestion
* Anypoint Connectors
```
See term presentation list, style reference.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
 
-image::amazon/amazon-dynamodb-create-table-use-case-flow.png[Creating a table in Amazon DynamoDB]
+* Amazon Web Services - DynamoDB
+* AWS Identity and Access Management (IAM) user account credentials to access AWS with the connector
```
Why qualify this list item but not the other two?
```suggestion
* AWS Identity and Access Management (IAM) user account credentials
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
+Before creating an app, you must have access to:
 
-image::amazon/amazon-dynamodb-create-table-use-case-flow.png[Creating a table in Amazon DynamoDB]
+* Amazon Web Services - DynamoDB
```
I looked this up and it seems that you can simply use...:
```suggestion
* Amazon DynamoDB
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector.adoc:
```diff
@@ -8,298 +8,58 @@ Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#
-----
-
-== Migration information
+Read the xref:amazon/amazon-dynamodb-connector-xml-maven.adoc[XML and Maven Support], and xref:amazon/amazon-dynamodb-connector-examples.adoc[Examples] topics.
```
no comma:
```suggestion
Read the xref:amazon/amazon-dynamodb-connector-xml-maven.adoc[XML and Maven Support] and xref:amazon/amazon-dynamodb-connector-examples.adoc[Examples] topics.
```

kbostrom reviewed (approved):
Edited

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+[[create-mule-project]]
+== Create a Mule Project
+
+In Studio, create a new Mule project in which to add and configure the connector.
```
GLOBALLY end text that directly introduces a list, steps, an example, a table, and so on with a colon:
```suggestion
In Studio, create a new Mule project in which to add and configure the connector:
```

dhirsch1 committed:
Updated a screenshot

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+In Studio, create a new Mule project in which to add and configure the connector.
+
+. In Anypoint Studio, from the main menu, select *File > New > Mule Project*.
+. Enter a name for your Mule project, and click *Finish*.
```
no comma necessary between two simple actions:
```suggestion
. Enter a name for your Mule project and click *Finish*.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+[[add-connector-to-project]]
+== Add the Connector to Your Mule Project
+
+Add Amazon DynamoDB Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as add the required dependencies to the project's `pom.xml` file.
```
```suggestion
Add Amazon DynamoDB Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as to add the required dependencies to the project's `pom.xml` file:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+Add Amazon DynamoDB Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as add the required dependencies to the project's `pom.xml` file.
+
+. In the *Mule Palette*, click *(X) Search in Exchange*.
```
```suggestion
. In *Mule Palette*, click *(X) Search in Exchange*.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+An input source initiates a flow when a specified condition is met. You can configure one of these input sources to use with Amazon DynamoDB Connector:
+
+* `HTTP Listener`, which initiates a flow each time it receives a request on the configured host and port.
```
Not sentences = no periods:
```suggestion
* `HTTP Listener`, which initiates a flow each time it receives a request on the configured host and port
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+An input source initiates a flow when a specified condition is met. You can configure one of these input sources to use with Amazon DynamoDB Connector:
+
+* `HTTP Listener`, which initiates a flow each time it receives a request on the configured host and port.
+* `Scheduler`, which initiates a flow when a time-based condition is met.
```
ditto:
```suggestion
* `Scheduler`, which initiates a flow when a time-based condition is met
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+For example, to configure `HTTP Listener`, follow these steps:
+
+. In the *Mule Palette*, select *HTTP*, and then select *Listener*.
```
```suggestion
. In *Mule Palette*, select *HTTP* > *Listener*.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+. Specify a value for the *Path* field.
+. Configure a global element that can be used by all instances of `HTTP Listener` in the app.
++
+To access the global element configuration fields, click the plus sign (+) next to the *Connector configuration* field.
```
This reads like a step?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+To add an operation for Amazon DynamoDB Connector, follow these steps:
+
+. In the *Mule Palette*, select *amazon dynamodb* and then select the desired operation.
```
```suggestion
. In *Mule Palette*, select *amazon dynamodb* and then select the desired operation.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+[[configure-global-element]]
+== Configure a Global Element for the Connector
+
+When you configure a connector, it is a good practice to configure a global element that can be used by all instances of that connector in the app. When configuring a global element, you must provide the authentication credentials the connector will use to access the target Amazon DynamoDB system. Amazon DynamoDB Connector supports both basic and role-based authentication.
```
Do we want to qualify this as a good practice or simply tell the audience to do this?
```suggestion
When you configure a connector, also configure a global element that can be used by all instances of that connector in the app. Configuring a global element requires you to provide the authentication credentials the connector uses to access the target Amazon DynamoDB system. Amazon DynamoDB Connector supports both basic and role-based authentication.
```

kbostrom replied:
Note the lowercasing of basic and role-based authentication; these are init. capped in the following task. 
Is that because we refer to their UI labels? If so, * *.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+. Select the connector in the Studio canvas.
+. In the configuration screen for the operation, click the plus sign (+) next to the `Connector configuration` field to access the global element configuration fields.
+. On the *General* tab, configure Basic or Role authentication, as described in the <<basic-authentication,Basic Authentication>> and <<role-authentication,Role Authentication>> sections.
```
See earlier question re. authentication names/UI labels.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+[[basic_authentication]]
+=== Basic Authentication
+
+Enter the following information on the *General* tab of the *Global Element Properties* screen to configure Basic authentication:
```
Note cap.
Standardize throughout...and would you please follow up with me so that I can update the Term Presentation List? THX

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+[%header,cols="30s,70a"]
+|===
+|Parameter |Description
```
These are User Actions more than Descriptions....

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+|Region Endpoint | Select the region endpoint for the service.
+|===
+
+For example:
```
Please replace these two words with a more precise fragment or sentence. See style ref.

dhirsch1 committed:
Interim commit

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+
+[%header,cols="30s,70a"]
+|===
+|Parameter |Description
```
ditto re. "Description" as a column heading

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+|Parameter |Description
+|Name |Enter the configuration name.
+|Connection | Select `Role`.
+|Role ARN | Enter the role to assume to gain cross account access.
```
```suggestion
|Role ARN | Enter the role to assume to gain cross-account access.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+|Region Endpoint | Select the region endpoint for the service.
+|===
+
+For example:
```
ditto re. more robust fragment or sentence, per style ref.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+[[configure-other-fields]]
+== Configure the Other Connector Fields
+
+After you configure a global element for Amazon DynamoDB connector, configure the the other required fields for the connector:
```
```suggestion
After you configure a global element for Amazon DynamoDB connector, configure the other required fields for the connector:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,155 @@
+|Field |Description
+|Table name | Name of the table to create
+|Attribute definitions | Attributes that describe the key schema for the table and its indexes
+|Key schemas | Attributes that make up the primary key for a table or index
```
```suggestion
|Key schemas | Attributes that compose the primary key for a table or index
```

kbostrom reviewed (approved):
Edited.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,75 @@
+include::_attributes.adoc[]
+endif::[]
+
+If you manually code a Mule runtime engine (Mule) app in XML either from the Anypoint Studio XML editor or from a text editor, you can get access to the connector in your app by adding reference statements to both your XML Mule flow and the Apache Maven `pom.xml` file.
```
```suggestion
If you manually code a Mule runtime engine (Mule) app in XML, either from the Anypoint Studio XML editor or from a text editor, you can get access to the connector in your app by adding reference statements to both your XML Mule flow and the Apache Maven `pom.xml` file.
```

kbostrom replied:
Every time I read this, I wonder why someone would want to do this manually instead of using Studio....

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,75 @@
+
+== Add a Namespace for the Connector
+
+Paste the following code inside the <mule> tag of the header of your configuration XML:
```
```suggestion
Paste the following code inside the `<mule>` tag of the header of your configuration XML:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,75 @@
+
+== Add a POM File Dependency
+
+The `pom.xml` Apache Maven file generated by Studio contains dependencies for a Mule app. When you code a Mule app manually, include this XML snippet in your `pom.xml` file to enable access to this connector:
```
```suggestion
The Apache Maven `pom.xml` file generated by Studio contains dependencies for a Mule app. When you code a Mule app manually, include this XML snippet in your `pom.xml` file to enable access to this connector:
```

kbostrom reviewed (approved):
Edited

dhirsch1 committed:
Some fixes and different steps for referencing the properties file

dhirsch1 committed:
Entered Kim's edits

dhirsch1 committed:
Commit of additional Studio edits

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,153 @@
+
+In Studio, create a new Mule project in which to add and configure the connector:
+
+. In Anypoint Studio, from the main menu, select *File > New > Mule Project*.
```
I know I've seen this a million times, but just now I wonder if we need "from the main menu"?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,153 @@
+[[add-connector-to-project]]
+== Add the Connector to Your Mule Project
+
+Add Amazon DynamoDB Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as to add the required dependencies to the project's `pom.xml` file:
```
"a" > "your"?
```suggestion
Add Amazon DynamoDB Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as to add the required dependencies to the project's `pom.xml` file:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,153 @@
+[[configure-global-element]]
+== Configure a Global Element for the Connector
+
+When you configure a connector, it is a good practice to configure a global element that can be used by all instances of that connector in the app. Configuring a global element requires you to provide the authentication credentials the connector uses to access the target Amazon DynamoDB system. Amazon DynamoDB Connector supports both Basic authentication and Role authentication.
```
Consideration:
I've been gathering info about the practice of (not) using "MuleSoft recommends"; alas, I lost my Notes in my laptop swap. However, I remember that Salesforce uses "it is best to," while I have been recommending instead that we simply document the best way to do something:
```suggestion
When you configure a connector, configure a global element that can be used by all instances of that connector in the app. Configuring a global element requires you to provide the authentication credentials the connector uses to access the target Amazon DynamoDB system. Amazon DynamoDB Connector supports both Basic authentication and Role authentication.
```

dhirsch1 replied:
Changing to "it is best to..."

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,153 @@
+. In the configuration screen for the operation, click the plus sign (+) next to the `Connector configuration` field to access the global element configuration fields.
+. On the *General* tab, configure Basic authentication or Role authentication, as described in the <<basic-authentication,Basic Authentication>> and <<role-authentication,Role Authentication>> sections.
++
+You can enter your authorization credentials in the global configuration properties or you can reference a configuration file that contains ANT-style property placeholders (recommended). For information, see xref:introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
```
Consideration: should we reverse the order of the methods of authorization so that referencing the config. file is first, since it is the preferred method?
Consider also the previous comment about recommendations.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-studio.adoc:
```diff
@@ -0,0 +1,153 @@
+== See Also
+
+* xref:introduction/introduction-to-anypoint-connectors.adoc[Introduction to Anypoint Connectors]
+* xref:introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]
```
Do we really need to reference the generic version of the same topic as this one?

kbostrom reviewed (approved):
Edited

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+include::_attributes.adoc[]
+endif::[]
+
+The following are common use cases for the Amazon DynamoDB connector:
```
```suggestion
The following are common use cases for the Amazon DynamoDB Connector:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure `HTTP Listener` to listen for HTTP requests on the `/createtable` path:
+
+. In the *Mule Palette*, search for `http` and select the *Listener* operation.
```
```suggestion
. In the *Mule Palette* view, search for `http` and select the *Listener* operation.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure the `Create Table` operation to create a table named `Issues`:
+
+. In the *Mule Palette*, click *Search in Exchange*, and search for `amazon dynamodb`.
```
```suggestion
. In the *Mule Palette* view, click *Search in Exchange* and search for `amazon dynamodb`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. Select *Amazon DynamoDB Connector*, click *Add>*, and then click *Finish*.
+. Drag the *Create Table* operation to the right of *Listener* on the Studio canvas.
+. Click the *Global Elements* link below the flow.
+. Add the properties file you created earlier to the *Configuration properties* field, as described in <<introduction/intro-connector-configuration-overview.adoc#property-placeholders,Use Property Placeholders for Property Values>>.
```
This looks like it should be documented as a Prerequisite. We want to avoid sending readers out of a task once started.

dhirsch1 replied:
This is not a prerequisite. It is documented in the intro to configuring connectors topic, which is why I am referring to it here.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+--
+[%header,cols="30s,40a,30a"]
+|===
+|Parameter |Description |Value
```
```suggestion
|Parameter |Description |User Action
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+|Name |Configuration name |Enter `create-table-flow`.
+|Access Key |Alphanumeric text string that uniquely identifies the user who owns the account |Enter the `${config.accesskey}` property placeholder.
+|Secret Key |Key that acts as a password |Enter the `${config.secretkey}` property placeholder.
+|Region Endpoint |Region for the Amazon DynamoDB client |`us-east-1`
```
Reword third-col entry to be parallel with others.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+* If the connection is successful, click *OK* to save the configuration.
+* If the connection is unsuccessful, review or correct any incorrect parameters, and then test again.
+. Click *OK*.
+. In the Create table properties window, configure the following fields:
```
Format whatever part of this is the UI label for the window; if none is, then lowercase "create"

dhirsch1 replied:
Not sure about this. Create table is an operation and this is a window for that operation.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
++
+[%header,cols="25,35,40a"]
+|===
+|Field |Description | Value
```
Ditto re. Value > User Action, as appropriate, GLOBALLY

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+* `Title` as key type `RANGE`
+|Read capacity units | The maximum number of strongly consistent reads consumed per second before Amazon DynamoDB returns a throttling exception | Enter `5`.
+|Write capacity units | The maximum number of writes consumed per second before Amazon DynamoDB returns a throttling exception | Enter `5`.
+|Global secondary indexes | An optional, additional key structure for the table | Enter `#[vars.secondary]` to create a variable for the secondary index, and then add the secondary index in the XML code. For information, see <<secondary-index, Add a Secondary Index to the Issues Table>>.
```
Reformat...but BETTER, move the xref to See Also to avoid sending the person doing the task off into the ether...:
```suggestion
|Global secondary indexes | An optional, additional key structure for the table | Enter `#[vars.secondary]` to create a variable for the secondary index, and then add the secondary index in the XML code. +
For information, see <<secondary-index, Add a Secondary Index to the Issues Table>>.
```
Should you decide to keep the xref in the table, try to be more specific about "for information." What kind of information, how related to this step?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+* The second item specifies the name of the table to create.
+* The third item defines the attributes of the new table.
+* The fourth item defines the key schemas of the new table.
+* The fifth items specify the maximum number of strongly consistent reads consumed per second and strongly consistent writes consumed per second before Amazon DynamoDB returns a throttling exception.
```
Two items in one list item for "fifth items"? A bit awk.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+[[secondary-index]]
+=== Add a Secondary Index to the Issues Table
+
+To add a global secondary index to the `Issues` table, add a variable before the `<dynamodb:create-table>` element in the XML.
```
Do we need to say anything about WHY our audience wants or needs to do this?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+You must include these elements in the XML:
+
+* `<Projection>`, which specifies the attributes to copy from the table into the index. In this example, the value `ALL` means that all attributes are copied.
```
Per our bullet-list item style...:
```suggestion
* `<Projection>`, which specifies the attributes to copy from the table into the index. +
In this example, the value `ALL` means that all attributes are copied.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure a `Logger` component to print to the Mule console the response generated by the `Create Table` operation:
+
+. In the *Mule Palette*, search for `logger`.
```
```suggestion
. In the *Mule Palette* view, search for `logger`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+. In the *Mule Palette*, search for `logger`.
+. Drag the *Logger* component to the right of *Create Table* on the Studio canvas.
+. Enter `#[payload]` in the *Message* field of the Logger properties window.
```
Format the name of the window as displayed in the UI, or lowercase "logger" if this is not a label.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+* The fifth items specify the maximum number of strongly consistent reads consumed per second and strongly consistent writes consumed per second before Amazon DynamoDB returns a throttling exception.
+* The sixth item species a variable for the secondary index.
+
+The XML flow for the Create Table example should look like this after you configure the `HTTP Listener` and `Create table` operations:
```
Not sure we should capitalize Create Table, since we never introduced it as a name....  GLOBAL

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+=== XML Flow for the Complete Create Table Example
+
+The XML flow for the complete Create Table example should look like this:
```
ditto re. example name? also, suggest omitting "should"

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Save and run the project as a Mule app:
+
+. In *Package Explorer*, right-click the project and click *Run As* > *Mule Application*.
```
```suggestion
. In *Package Explorer*, right-click the project name and click *Run As* > *Mule Application*.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+[[delete-table]]
+== Delete an Amazon DynamoDB Table
+
+This example shows how to use Amazon DynamoDB connector to delete an Amazon DynamoDB table.
```
Anything to say about under what circumstances our audience wants or needs to do this?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure `HTTP Listener` to listen for HTTP requests on the `/deletetable` path:
+
+. In the *Mule Palette*, search for `http` and select the *Listener* operation.
```
```suggestion
. In the *Mule Palette* view, search for `http` and select the *Listener* operation.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. Drag the *Listener* operation onto the Studio canvas.
+. In the Listener properties window, click the plus sign (+) next to the *Connector configuration* field to access the global element configuration fields for `HTTP Listener`.
+. In the *Host* field, select `localhost` and click *OK*.
+. In the *Path* field on the Listener properties window, enter `/deletetable`.
```
Do we need to repeat "on the Listener properties window"? They haven't left, right?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure the `Delete table` operation to delete the `Issues` table.
+
+. In the *Mule Palette*, click *Search in Exchange* and search for `amazon dynamodb`.
```
```suggestion
. In the *Mule Palette* window, click *Search in Exchange* and search for `amazon dynamodb`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. Select *Amazon DynamoDB Connector*, click *Add>*, and then click *Finish*.
+. Drag the *Delete Table* operation to the right of *Listener* on the Studio canvas.
+. Click the *Global Elements* link below the flow.
+. Add the properties file you created earlier to the *Configuration properties* field, as described in <<introduction/intro-connector-configuration-overview.adoc#property-placeholders,Use Property Placeholders for Property Values>>.
```
ditto re. sending the user out to another topic; omit or move to See Also or prerequisties

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+|Region Endpoint |Region for the Amazon DynamoDB client |`us-east-1`
+|===
++
+The corresponding XML configuration should be as follows:
```
```suggestion
The corresponding XML configuration looks like this:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+* If the connection is successful, click *OK* to save the configuration.
+* If the connection is unsuccessful, review or correct any incorrect parameters, and then test again.
+. Click *OK*.
+. In the Delete table properties window, set the value of `Table name` to `Issues`:
```
lowercase or format as a UI label

kbostrom replied:
```suggestion
. In the Delete table properties window, set the value of `Table name` to `Issues`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure a `Logger` component to print to the Mule console the response generated by the `Delete table` operation:
+
+. In the *Mule Palette*, search for `logger`.
```
```suggestion
. In the *Mule Palette* view, search for `logger`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+. In the *Mule Palette*, search for `logger`.
+. Drag the *Logger* component to the right of *Delete Table* on the Studio canvas.
+. In the Logger properties window, enter `#[payload]` in the *Message* field.
```
ditto prev. about window name

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+=== XML Flow for the Delete Table Example
+
+The XML flow for the Delete Table example should look like this:
```
```suggestion
The XML flow for the Delete Table example looks like this:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Save and run the project as a Mule app:
+
+. In *Package Explorer*, right-click the project and click *Run As* > *Mule Application*.
```
```suggestion
. In *Package Explorer*, right-click the project name and click *Run As* > *Mule Application*.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. In *Package Explorer*, right-click the project and click *Run As* > *Mule Application*.
+. Open a browser and check the response after you enter the `+http://localhost:8081/deletetable+` URL.
++
+You should see the generated response from the `Delete table` operation in the Mule console.
```
```suggestion
You see the generated response from the `Delete table` operation in the Mule console.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+This example shows how to use Amazon DynamoDB Connector to scan an Amazon DynamoDB table.
+
+The following considerations apply to performing a scan using Amazon DynamoDB Connector:
```
Are these considerations or Prerequisites?

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+The following considerations apply to performing a scan using Amazon DynamoDB Connector:
+
+* The DynamoDB AWS SDK paginates results by dividing them into specific pages. A one MB limit applies to returned results. When this limit is exceeded, you must perform another scan to receive the rest of the data.
```
One sentence max, GLOBALLY:
```suggestion
* The DynamoDB AWS SDK paginates results by dividing them into specific pages. +
A 1-MB limit applies to returned results. When this limit is exceeded, you must perform another scan to receive the rest of the data.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+The following considerations apply to performing a scan using Amazon DynamoDB Connector:
+
+* The DynamoDB AWS SDK paginates results by dividing them into specific pages. A one MB limit applies to returned results. When this limit is exceeded, you must perform another scan to receive the rest of the data.
+* To perform subsequent scans, apply the value of the `LastEvaluatedKey` field in the JSON response to the `Exclusive start keys` field in the Studio configuration of the `Scan` operation. When the operation has returned all pages of data, the `LastEvaluatedKey` value in the JSON response is null.
```
ditto
```suggestion
* To perform subsequent scans, apply the value of the `LastEvaluatedKey` field in the JSON response to the `Exclusive start keys` field in the Studio configuration of the `Scan` operation. +
When the operation has returned all pages of data, the `LastEvaluatedKey` value in the JSON response is null.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+Configure `HTTP Listener` to listen for HTTP requests on the `/scantable` path:
+
+. In the *Mule Palette*, search for `http` and select the *Listener* operation.
```
```suggestion
. In the *Mule Palette* view, search for `http` and select the *Listener* operation.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+. In the *Mule Palette*, search for `http` and select the *Listener* operation.
+. Drag the *Listener* operation onto the Studio canvas.
+. In the Listener properties window, click the plus sign (+) next to the *Connector configuration* field to access the global element configuration fields for `HTTP Listener`.
```
ditto GLOBAL re. window names. Please search for and evaluate all text around "window" per previous edits. I will stop marking.

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+=== Configure the Scan Operation
+
+. In the *Mule Palette*, click *Search in Exchange* and search for `amazon dynamodb`.
```
GLOBAL re. "view"; please search for and fix. I'll not continue marking it.
```suggestion
. In the *Mule Palette* view, click *Search in Exchange* and search for `amazon dynamodb`.
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. Select *Amazon DynamoDB Connector*, click *Add>*, and then click *Finish*.
+. Drag the *Scan* operation to the right of *Listener* on the Studio canvas.
+. Click the *Global Elements* link below the flow.
+. Add the properties file you created earlier to the *Configuration properties* field, as described in <<introduction/intro-connector-configuration-overview.adoc#property-placeholders,Use Property Placeholders for Property Values>>.
```
ditto prev. edits to this text earlier

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+
+=== XML for the Scan Table Example
+
+The XML for the Scan Table example should look like this:
```
```suggestion
The XML for the Scan Table example looks like this:
```

kbostrom commented on modules/ROOT/pages/amazon/amazon-dynamodb-connector-examples.adoc:
```diff
@@ -0,0 +1,700 @@
+. In *Package Explorer*, right-click the project and click *Run As* > *Mule Application*.
+. Open a browser and check the response after you enter the `+http://localhost:8081/scantable+` URL.
++
+You should see the generated response from the `Scan` operation in the Mule console.
```
```suggestion
You see the generated response from the `Scan` operation in the Mule console.
```

kbostrom replied:
GLOBALLY avoid modal verbs; see style ref.
GLOBALLY add "view" when preceding *Mule Palette* with "the."
GLOBALLY check with names and example names.

kbostrom reviewed (approved):
Done

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Finished entering doc edits
